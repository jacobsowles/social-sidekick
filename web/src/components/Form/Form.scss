.form {
  width: 100%;

  .form-buttons {
    margin-top: 10px;

    button {
      margin-right: 10px;

      &:last-child {
        margin-right: 0;
      }
    }
  }

  .form-field {
    display: block;

    input {
      width: calc(100% - 40px);
    }

    svg {
      opacity: 0.8;
      width: 40px !important;
    }

    textarea {
      width: 100%;
    }
  }
}
