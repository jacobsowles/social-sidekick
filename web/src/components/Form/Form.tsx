import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import './Form.scss';

//// FIELD ////

interface FieldProps {
  className?: string;
}

const Field: FunctionComponent<FieldProps> = ({ children, className }) => {
  return <div className={classNames('form-field', className)}>{children}</div>;
};

Field.displayName = 'Form.Field';

//// WRAPPER ////

interface WrapperProps {
  className?: string;
  onSubmit?: (event: any) => void;
}

const Wrapper: FunctionComponent<WrapperProps> = ({ children, className, onSubmit }) => {
  return (
    <form className={classNames('form', className)} onSubmit={onSubmit}>
      {children}
    </form>
  );
};

Wrapper.displayName = 'Form.Wrapper';

export default { Field, Wrapper };
