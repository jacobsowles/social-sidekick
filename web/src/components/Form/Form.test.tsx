import React from 'react';
import { init, shallow, toJson } from '@tests/component-test-base';
import Form from './Form';

describe('Form', () => {
  init(
    () => (
      <Form.Wrapper>
        <Form.Field>Test</Form.Field>
      </Form.Wrapper>
    ),
    'div'
  );

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
