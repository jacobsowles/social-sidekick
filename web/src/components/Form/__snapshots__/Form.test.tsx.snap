// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render the component without crashing 1`] = `
<form
  className="form"
>
  <Form.Field>
    Test
  </Form.Field>
</form>
`;
